prime-dotnet-webapi/Services/AgreementService.cs

```diff
@@ -278,6 +278,18 @@ public async Task<SignedAgreementDocument> AddSignedAgreementDocumentAsync(int a
             return signedAgreement;
         }
 
+        public async Task<IEnumerable<AgreementVersionViewModel>> GetLatestEnrolleeAgreementVersionsAsync()
+        {
+            var agreementVersionList = new List<AgreementVersion>();
+            foreach (var type in AgreementTypeExtensions.EnrolleeAgreementTypes())
+            {
+                agreementVersionList.Add(await FetchNewestAgreementVersionOfTypeAsync(type));
```

Comment:
As an FYI, this is going to hit the DB once for every AgreementType

Reply:
> As an FYI, this is going to hit the DB once for every AgreementType

yep aware of that. Thought about one single query but seems linq is limited to achieve it.